src/contexts/current-product.tsx

```diff
@@ -1,17 +1,22 @@
-import { createContext, FC, useContext } from 'react'
+import { createContext, ReactNode, useContext } from 'react'
 import { useRouter } from 'next/router'
-import { Product } from 'types/products'
+import { ProductData } from 'types/products'
 
 export type RouteChangeStartHandler = (url: string) => void
 
-type CurrentProduct = Product | undefined
+type CurrentProduct = ProductData | undefined
 
 const CurrentProductContext = createContext<CurrentProduct>(undefined)
 
-const CurrentProductProvider: FC<{ currentProduct: Product }> = ({
+interface CurrentProductProviderProps {
+  children: ReactNode
```

Comment:
Went with this approach instead of the previously used `React.FC` approach because there was a typedef warning about the props not being typed.